src/cmd/troubleshoot/image_pullable.go

```diff
@@ -0,0 +1,300 @@
+
+	pullSecret, err := getPullSecretToken(troubleshootCtx)
+	if err != nil {
+		return err
+	}
+
+	dynakubeOneAgentImage := getOneAgentImageEndpoint(troubleshootCtx)
+	if err != nil {
+		return err
+	}
+
+	if err = checkComponentImagePullable(troubleshootCtx.httpClient, "OneAgent", pullSecret, dynakubeOneAgentImage); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func checkActiveGateImagePullable(troubleshootCtx *troubleshootContext) error {
+	logNewTestf("checking if ActiveGate image is pullable ...")
+
+	pullSecret, err := getPullSecretToken(troubleshootCtx)
+	if err != nil {
+		return err
+	}
+
+	dynakubeActiveGateImage := getActiveGateImageEndpoint(troubleshootCtx)
+	if err != nil {
+		return err
+	}
+
+	if err = checkComponentImagePullable(troubleshootCtx.httpClient, "ActiveGate", pullSecret, dynakubeActiveGateImage); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func checkComponentImagePullable(httpClient *http.Client, componentName string, pullSecret string, componentImage string) error {
+	// split activegate image into registry and image name
+	componentRegistry, componentImage, componentVersion, err := splitImageName(componentImage)
+	if err != nil {
+		return err
+	}
+	logInfof("using '%s' on '%s' with version '%s' as %s image", componentImage, componentRegistry, componentVersion, componentName)
+
+	imageWorks := false
+
+	// parse docker config
+	var result Auths
+	json.Unmarshal([]byte(pullSecret), &result)
+
+	for registry, endpoint := range result.Auths {
+		logInfof("checking images for registry '%s'", registry)
+
+		apiToken := base64.StdEncoding.EncodeToString([]byte(endpoint.Username + ":" + endpoint.Password))
+
+		if statusCode, err := connectToDockerRegistry(httpClient, "HEAD", "https://"+registry+"/v2/", "Basic", apiToken); err != nil {
+			logErrorf("registry '%s' unreachable", registry)
+			continue
+		} else {
+			if statusCode != 200 {
+				logErrorf("registry '%s' unreachable (%d)", registry, statusCode)
+				continue
+			} else {
+				logInfof("registry '%s' is accessible", registry)
+			}
+		}
+
+		if statusCode, err := connectToDockerRegistry(httpClient, "HEAD", "https://"+registry+"/v2/"+componentImage+"/manifests/"+componentVersion, "Basic", apiToken); err != nil {
+			logErrorf("registry '%s' unreachable", registry)
+			continue
+		} else {
+			if statusCode != 200 {
+				logErrorf("image '%s' with version '%s' not found on registry '%s'", componentImage, componentVersion, registry)
+				continue
+			} else {
+				logInfof("image '%s' with version '%s' exists on registry '%s", componentImage, componentVersion, registry)
+			}
+		}
+
+		imageWorks = true
+	}
+
+	if imageWorks {
+		logOkf("%s image '%s' found", componentName, componentRegistry+"/"+componentImage)
+	} else {
+		logErrorf("%s image '%s' missing", componentName, componentRegistry+"/"+componentImage)
+		return fmt.Errorf("%s image '%s' missing", componentName, componentRegistry+"/"+componentImage)
```

Comment:
we should build the error message only once and reuse it

Reply:
redundant logError calls removed